### PR TITLE
refactor: assessment

### PR DESCRIPTION
**Pull request by yash6102000**
optimize assessment route

### CODE REVIEW

File: app/components/tests/TestList.tsx [hunk 1]
```diff
@@ -31,6 +31,7 @@ const TestList = () => {
   //loader and action data
   const testLoaderData = useLoaderData()
   const testActionData = useActionData()
+  console.log(testActionData, "testActionData")
```
Comment (Dhruv-Copods) [inline app/components/tests/TestList.tsx:34]: Remove console log

File: app/services/assessments.service.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    testsItemsPerPage,
+    testsCurrentPage
+  )
+    .then((res) => {
+      return cb(res, "statusCheck.success")
+    })
+    .catch((err) => {
+      return cb(err, "")
+    })
+}
```
Comment (Dhruv-Copods) [inline app/services/assessments.service.ts:54]: It is a server call we should use Async Await

File: app/services/assessments.service.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+/** function for deleting the assessment by id
+ * @param id
+ * @returns this function will delete the assessment by id
+ */
+
+export const deleteAssessmentById = async (id: string) => {
+  const deletedHandle = deleteTestById(id)
```
Comment (Dhruv-Copods) [inline app/services/assessments.service.ts:62]: Add Await

File: app/routes/$workspaceId/assessments/$testId.tsx [hunk 4]
```diff
@@ -1,23 +1,25 @@
...
 import type { ActionFunction, LoaderFunction } from "@remix-run/server-runtime"
 import invariant from "tiny-invariant"
-
 import TestDetails from "~/components/tests/TestDetails"
-import { createCandidate } from "~/models/candidate.server"
 import { getTestById } from "~/models/tests.server"
 import { getUserWorkspaces } from "~/models/workspace.server"
-import { getUserId, requireUserId } from "~/session.server"
+import {
+  createCandidateByAssessId,
```
Comment (Dhruv-Copods) [inline app/routes/$workspaceId/assessments/$testId.tsx:8]: Think of better naming for this function

File: app/routes/$workspaceId/assessments/$testId.tsx [hunk 5]
```diff
@@ -1,23 +1,25 @@
...
+  getRequiredUserId,
+  getUsersId,
+  getWorkspaces,
+} from "~/services/assessments.service"
 
 type LoaderData = {
   testPreview: Awaited<ReturnType<typeof getTestById>>
   workspaces: Awaited<ReturnType<typeof getUserWorkspaces>>
   currentWorkspaceId: string
```
Comment (Dhruv-Copods) [inline app/routes/$workspaceId/assessments/$testId.tsx:17]: These Return Types need to be changed with new functions

File: app/routes/$workspaceId/assessments/$testId.tsx [hunk 6]
```diff
@@ -1,23 +1,25 @@
...
 }
-
 export const loader: LoaderFunction = async ({ request, params }) => {
-  const userId = await getUserId(request)
+  const userId = await getUsersId(request)
   const currentWorkspaceId = params.workspaceId as string
-  const workspaces = await getUserWorkspaces(userId as string)
+  const workspaces = await getWorkspaces(userId as unknown as string)
```
Comment (Dhruv-Copods) [inline app/routes/$workspaceId/assessments/$testId.tsx:22]: unknown is not required here if we return the function form service

File: app/models/tests.server.ts [hunk 7]
```diff
@@ -1,8 +1,6 @@
-import type { Section } from "@prisma/client"
-
 import { prisma } from "~/db.server"
 
-export async function getTestById({ id }: Pick<Section, "id">) {
+export async function getTestById(id: string) {
```
Comment (anuragP04) [inline app/models/tests.server.ts:3]: why did we have this change?

File: app/routes/$workspaceId/assessments/$testId.tsx [hunk 8]
```diff
@@ -1,25 +1,25 @@
...
+  testPreview: Awaited<ReturnType<typeof getAssessmentById>>
+  workspaces: Awaited<ReturnType<typeof getWorkspaces>>
   currentWorkspaceId: string
 }
-
 export const loader: LoaderFunction = async ({ request, params }) => {
   const userId = await getUserId(request)
   const currentWorkspaceId = params.workspaceId as string
-  const workspaces = await getUserWorkspaces(userId as string)
+  const workspaces = await getWorkspaces(userId as unknown as string)
```
Comment (anuragP04) [inline app/routes/$workspaceId/assessments/$testId.tsx:21]: remove unknown type

File: app/models/tests.server.ts (hunk 7 above)
Comment (yash6102000) [reply app/models/tests.server.ts:3]: beacuse , where this wrapper used, i passed the id in string from and this function is inside the wrapper that's why

File: app/routes/$workspaceId/assessments/$testId.tsx [hunk 9]
```diff
@@ -1,25 +1,25 @@
...
   invariant(params.testId, "testId not found")
-  const testPreview = await getTestById({ id: params.testId })
+  const testPreview = await getAssessmentById(params.testId)
```
Comment (anuragP04) [inline app/routes/$workspaceId/assessments/$testId.tsx:23]: don't update the params type

File: app/routes/$workspaceId/assessments/$testId.tsx [hunk 10]
```diff
@@ -46,11 +47,11 @@ export const action: ActionFunction = async ({ request }) => {
         testId,
       })
     }
-    const candidateInviteStatus = await createCandidate({
+    const candidateInviteStatus = await getCandidateByAssessmentId(
```
Comment (anuragP04) [inline app/routes/$workspaceId/assessments/$testId.tsx:51]: don't update the argument type

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**